src/sequence/SequencePipe.h

```diff
@@ -131,14 +131,19 @@ class SequencePipe : public Pipe {
       for (int k = 0; k < unigrams.size(); ++k) {
         int r = unigrams[k];
         if (!NEARLY_EQ_TOL(gold_outputs[r], predicted_outputs[r], 1e-6)) {
-          ++num_tag_mistakes_;
```

Comment:
I think we may assume that evaluation will never be performed with multithreading turned on. I'd rather abstract away the threading functions from everything else, moving them to a layer above.